### ISSUE TITLE
patch request body having attributes with null value are not sent to server

### ISSUE THREAD

**Issue by devstartshop** [lagoshny/ngx-hal-client#13]
file **resource-helper.ts** has a method **resolveRelations** which is skipping the attributes having null or undefined values.

My use case is that I need to set the expiry date of a offer as null in database. The offer in database has an expiry date set and we need to extend the offer indefinitely.

Kindly review this and suggest fix or workaround of this issue.

**Comment by lagoshny** [comment]
Hi @devstartshop , I think that we not need to add null values to request body, because by default your server side (Spring app) accepts non present values as null values.

Can you provide more information about your system communication between client side and database? Is it doing through Spring app? Or whatever? And may be you have additional Jackson mapper settings for your dto objects?

**Comment by devstartshop** [comment]
Hi @lagoshny, my app is using angular9 with spring data rest. 

I am using patch calls for making partial updates to my entity. Kindly suggest how to unset a field in my backend entity. I want to set the date field as null in database.

**Comment by lagoshny** [comment]
I understand your problem.
Unfortunately, this is not possible to do at this time.
I will fix it today or tomorrow.

**Comment by lagoshny** [comment]
@devstartshop ,
I added feature that allows pass property null values to JSON request.
It's available in **1.0.20** version.

You have two options:

1) Include all entity null values from your entity:
```
yourService.patch(someEntity, Include.NULL_VALUES)
``` 
2) Include only specific entity null values:
```
yourService.patch(someEntity, [{props: ['propName1', 'propName2', ... , 'propNameN'], include: Include.NULL_VALUES}])
``` 

See more details in [CHANGELOG](https://github.com/lagoshny/ngx-hal-client/blob/master/CHANGELOG.md#1020-2020-04-27)

**Comment by devstartshop** [comment]
@lagoshny - I am getting below error after upgrading to 1.0.20 version:
`
Uncaught ReferenceError: Cannot access 'ExternalService' before initialization
    at Module../node_modules/@lagoshny/ngx-hal-client/fesm2015/lagoshny-ngx-hal-client.js (lagoshny-ngx-hal-client.js:780)
    at __webpack_require__ (bootstrap:84)
`

I am working with angular 9.1.6 and ivy turned off.

**Comment by lagoshny** [comment]
@devstartshop , I fixed it in **1.0.22** version, please get update to this version.

**Comment by devstartshop** [comment]
> @devstartshop , I fixed it in **1.0.22** version, please get update to this version.

Thanks @lagoshny . I am trying this right now. Will update you soon.

**Comment by devstartshop** [comment]
@lagoshny - I tested this and its working fine for me.

**Comment by devstartshop** [comment]
> > @devstartshop , I fixed it in **1.0.22** version, please get update to this version.
> 
> Thanks @lagoshny . I am trying this right now. Will update you soon.

@lagoshny - I tested this and its working fine. Thanks for providing a solution to this.
